res/res_pjsip_caller_id.c

```diff
@@ -35,6 +35,25 @@
 #include "asterisk/callerid.h"
 #include "asterisk/conversions.h"
 
+static int extract_oli(const pjsip_param *param_list, char *oli, size_t len)
+{
+	static const pj_str_t oli_str1 = { "isup-oli", 8 };
+	static const pj_str_t oli_str2 = { "ss7-oli", 7 };
+	static const pj_str_t oli_str3 = { "oli", 3 };
+	pjsip_param *oli1, *oli2, *oli3;
```

Comment:
You don't need 3 separate pointers for these, you can just re-use one of them in each `if` branch.

You could take that a step further (which I like but may not be everyone's cup of tea):

```
static int extract_oli(const pjsip_param *param_list, char *oli, size_t len)
{
    static const pj_str_t oli_str1 = { "isup-oli", 8 };
    static const pj_str_t oli_str2 = { "ss7-oli", 7 };
    static const pj_str_t oli_str3 = { "oli", 3 };
    pjsip_param *ov;

    if ((ov = pjsip_param_find(param_list, &oli_str1))
        || (ov = pjsip_param_find(param_list, &oli_str2))
        || (ov = pjsip_param_find(param_list, &oli_str3))) {
        ast_copy_pj_str(oli, &ov->value, len);
        return 0;
    }

    return -1;
}
```

Reply:
For clarity am requesting you re-use a single pointer. The second bit is optional.